File: php/example_code/ses/Create_Rule.php
<?php
/**
 * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * This file is licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License. A copy of
 * the License is located at
 *
 * http://aws.amazon.com/apache2.0/
 *
 * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * ABOUT THIS PHP SAMPLE => This sample is part of the SDK for PHP Developer Guide topic at
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/ses-filters.html
 *
 */
// snippet-start:[ses.php.create_rule.complete]
// snippet-start:[ses.php.create_rule.import]

require 'vendor/autoload.php';

use Aws\SES\SESClient; 
use Aws\Exception\AwsException;
// snippet-end:[ses.php.create_rule.import]

//Create a SESClient 
// snippet-start:[ses.php.create_rule.main]
$SesClient = new Aws\SES\SESClient([
    'profile' => 'default',
    'version' => '2010-12-01',
    'region' => 'us-east-2'
]);

$rule_name = 'Rule_Name';
$rule_set_name = 'Rule_Set_Name';
$s3_bucket = 'Bucket_Name';

try {
    $result = $SesClient->createReceiptRule([
        'Rule' => [
            'Actions' => [
                [
                    'S3Action' => [
                        'BucketName' => $s3_bucket,
                    ],
                ],
            ],
            'Name' => $rule_name,
            'ScanEnabled' => true,
            'TlsPolicy' => 'Optional',
            'Recipients' => ['<string>', ...]
        ],
        'RuleSetName' =>  $rule_set_name,
        
     ]);
    var_dump($result);
} catch (AwsException $e) {
    // output error message if fails
    echo $e->getMessage();
    echo "\n";
}
 
 
// snippet-end:[ses.php.create_rule.main]
// snippet-end:[ses.php.create_rule.complete]
// snippet-comment:[These are tags for the AWS doc team's sample catalog. Do not remove.]
// snippet-sourcedescription:[Create_Rule.php demonstrates how to control your incoming email by adding a receipt rule to an existing receipt rule set. This example shows you how to create a receipt rule that sends incoming messages to an Amazon S3 bucket, but you can also send messages to Amazon SNS, and AWS Lambda.]
// snippet-keyword:[PHP]
// snippet-keyword:[AWS SDK for PHP v3]
// snippet-keyword:[Code Sample]
// snippet-keyword:[Amazon Simple Email Service]
// snippet-service:[ses]
// snippet-sourcetype:[full-example]
// snippet-sourcedate:[2018-09-20]
// snippet-sourceauthor:[jschwarzwalder (AWS)]

File: go/cloudwatch/CreateRule/CreateRule.go
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.

   This file is licensed under the Apache License, Version 2.0 (the "License").
   You may not use this file except in compliance with the License. A copy of
   the License is located at

    http://aws.amazon.com/apache2.0/

   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
   CONDITIONS OF ANY KIND, either express or implied. See the License for the
   specific language governing permissions and limitations under the License.
*/
// snippet-start:[cloudwatch.go.create_rule]
package main

// snippet-start:[cloudwatch.go.create_rule.imports]
import (
    "flag"
    "fmt"

    "github.com/aws/aws-sdk-go/aws/session"
    "github.com/aws/aws-sdk-go/service/cloudwatchevents"
)
// snippet-end:[cloudwatch.go.create_rule.imports]

// CreateRule creates a rule that watches for events on a schedule.
// Inputs:
//     sess is the current session, which provides configuration for the SDK's service clients
//     ruleName is the name of the rule
//     roleARN is the ARN of the role
//     schedule is the schedule expression
// Output:
//     If successful, a PutRuleOutput and nil
//     Otherwise, nil and the error from a call to PutRule
func CreateRule(sess *session.Session, ruleName *string, roleARN *string, schedule *string) (*cloudwatchevents.PutRuleOutput, error) {
    // Create the service client
    // snippet-start:[cloudwatch.go.create_rule.call]
    svc := cloudwatchevents.New(sess)

    result, err := svc.PutRule(&cloudwatchevents.PutRuleInput{
        Name:               ruleName,
        RoleArn:            roleARN,
        ScheduleExpression: schedule,
    })
    // snippet-end:[cloudwatch.go.create_rule.call]
    if err != nil {
        return nil, err
    }

    return result, nil
}

func main() {
    // snippet-start:[cloudwatch.go.create_rule.args]
    ruleName := flag.String("r", "", "The name of the rule")
    roleARN := flag.String("a", "", "The ARN of the role")
    schedule := flag.String("s", "", "The schedule expression")
    flag.Parse()

    if *ruleName == "" || *roleARN == "" || *schedule == "" {
        fmt.Println("You must supply a rule name (-r RULE), role ARN (-a ROLE-ARN), and schedule expression (-s EXPRESSION)")
        return
    }
    // snippet-end:[cloudwatch.go.create_rule.args]

    // Initialize a session that the SDK uses to load
    // credentials from the shared credentials file (~/.aws/credentials)
    // snippet-start:[cloudwatch.go.create_rule.session]
    sess := session.Must(session.NewSessionWithOptions(session.Options{
        SharedConfigState: session.SharedConfigEnable,
    }))
    // snippet-end:[cloudwatch.go.create_rule.session]

    result, err := CreateRule(sess, ruleName, roleARN, schedule)
    if err != nil {
        fmt.Println("Got an error creating the rule:")
        fmt.Println(err)
        return
    }

    // snippet-start:[cloudwatch.go.create_rule.print]
    fmt.Println("Rule ARN:" + *result.RuleArn)
    // snippet-end:[cloudwatch.go.create_rule.print]
}
// snippet-end:[cloudwatch.go.create_rule]